beacon-chain/sync/initial-sync/sync_blocks.go

```diff
@@ -0,0 +1,172 @@
+package initialsync
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/prysmaticlabs/prysm/shared/p2p"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"go.opencensus.io/trace"
+)
+
+func (s *InitialSync) processBlockAnnounce(msg p2p.Message) {
+	_, span := trace.StartSpan(msg.Ctx, "beacon-chain.sync.initial-sync.processBlockAnnounce")
+	defer span.End()
+	data := msg.Data.(*pb.BeaconBlockAnnounce)
+	recBlockAnnounce.Inc()
+
+	if s.stateReceived && data.SlotNumber > s.highestObservedSlot {
```

Comment:
If we have a block that is higher than the highest observed one, we send a batch request for it